src/main/java/com/swirlds/recordserver/TokensService.java

```diff
@@ -143,4 +146,121 @@ private void getDefaultMessageHandler(ServerRequest request, ServerResponse resp
+        if (resultTableResultSet.getRowCount() > 1) {
+            returnObject.add("count", resultTableResultSet.getRowCount());
+        }
+
+        final long consensusTime = resultTableResultSet.getLong(0, 0);
+        final JsonObject fields = parseFromColumn(resultTableResultSet.getString(0, 6));
+
+        addIfKeyNotNull(returnObject, "admin_key", fields, "adminKey");
+        returnObject.add("alias", resultTableResultSet.getString(0, 3));
+        addIfNotNull(returnObject, "auto_renew_account", fields.getString("autoRenewAccount", null));
+        addIfNotNull(returnObject, "auto_renew_period", fields.getString("autoRenewPeriod", null));
+        returnObject.add("created_timestamp", consensusTime);
+        // no data retained for "deleted" (boolean) field
+        addIfNotNull(returnObject, "decimals", fields.getJsonNumber("decimals"));
+        returnObject.add("evm_address", resultTableResultSet.getString(0, 2));
+        addIfNotNull(returnObject, "expiry_timestamp", fields.getJsonNumber("expiry"));
+        addIfNotNull(returnObject, "freeze_default", fields.getBoolean("freezeDefault", false));
+        addIfKeyNotNull(returnObject, "freeze_key", fields, "freezeKey");
+        addIfNotNull(returnObject, "initial_supply", fields.getString("initialSupply", null));
+        addIfKeyNotNull(returnObject, "kyc_key", fields, "kycKey");
+        addIfNotNull(returnObject, "max_supply", fields.getString("maxSupply", null));
+        addIfNotNull(returnObject, "memo", fields.getString("memo", null));
+        returnObject.add("modified_timestamp", consensusTime);
+        addIfNotNull(returnObject, "name", fields.getString("name", null));
+        addIfKeyNotNull(returnObject, "pause_key", fields, "pauseKey");
+        // no data retained for pause_status field
+        addIfNotNull(returnObject, "public_key", resultTableResultSet.getString(0, 5));
+        addIfNotNull(returnObject, "public_key_type", resultTableResultSet.getString(0, 4));
+        addIfNotNull(returnObject, "realm", fields.getString("realm", null));
+        addIfNotNull(returnObject, "shard", fields.getString("shard", null));
+        addIfKeyNotNull(returnObject, "supply_key", fields, "supplyKey");
+        addIfNotNull(returnObject, "supply_type", fields.getString("supplyType", null));
+        addIfNotNull(returnObject, "symbol", fields.getString("symbol", null));
+        returnObject.add("token_id", resultTableResultSet.getLong(0, 1));
+        // no data retained for total_supply field
+        addIfNotNull(returnObject, "treasury_account_id", fields.getString("treasury", null));
+        addIfNotNull(returnObject, "type", fields.getString("tokenType", null));
+        addIfKeyNotNull(returnObject, "wipe_key", fields, "wipeKey");
+        // no data retained for custom_fees field
+
+        response.send(returnObject.build());
+    }
+
+    private void addIfNotNull(JsonObjectBuilder object, String name, String value) {
```

Comment:
I actually like this and I think we should use this consistently in all our endpoints. Was wondering if we can move it to a utils class or something.

Reply:
I am perfectly fine moving this; I was also thinking of maybe updating the `listTokens()` routine (same `TokenServices.java` file) to use it as well.  As far as existing utilities classes -- we have a "**utils**" directory with a `Utils.java` that has routines like toHex() and toBase64() that I think would be a good place to put these.  Or I could create a new utility class either in that directory, or just inside the `src/main/java/com/swirlds/recordserver` directory, if you prefer.